dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/setup/ExecutorServiceBuilder.java

```diff
@@ -48,12 +50,12 @@ public ExecutorServiceBuilder(LifecycleEnvironment environment, String nameForma
         this(environment, nameFormat, buildThreadFactory(nameFormat));
     }
 
+    @SuppressWarnings("ResultOfMethodCallIgnored")
     private static ThreadFactory buildThreadFactory(String nameFormat) {
+        String.format(Locale.ROOT, nameFormat, 0); // Fail fast on invalid name format
         return r -> {
             final Thread thread = Executors.defaultThreadFactory().newThread(r);
-            if (nameFormat != null) {
```

Comment:
Not sure why this was relaxed in 2.x to allow null nameFormat. Previously, Guava validated the nameFormat was of a valid format and didn't allow null. If this is required, I can make it able to handle a null nameFormat.